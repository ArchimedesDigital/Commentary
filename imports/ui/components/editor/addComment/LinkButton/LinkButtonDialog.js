import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import TextField from 'material-ui/TextField';

const LinkButtonDialog = ({ open, linkValue, handleClose, handleAddLink, handleRemoveLink, onValueChange }) => {
	const actions = [
		<FlatButton
			label="Cancel"
			primary
			onClick={handleClose}
		/>,
		<FlatButton
			label="Remove link"
			primary
			onClick={handleRemoveLink}
		/>,
		<FlatButton
			label="Add link"
			primary
			onClick={handleAddLink}
		/>,
	];

	return (
		<Dialog
			title="Add link"
			actions={actions}
			open={open}
			onRequestClose={handleClose}
		>
			<TextField
				hintText="URL"
				fullWidth
				defaultValue={linkValue}
				onChange={onValueChange}
			/>
		</Dialog>
	);
};
LinkButtonDialog.propTypes = {
	open: React.PropTypes.bool.isRequired,
	linkValue: React.PropTypes.string,
	handleClose: React.PropTypes.func.isRequired,
	handleAddLink: React.PropTypes.func.isRequired,
	handleRemoveLink: React.PropTypes.func.isRequired,
	onValueChange: React.PropTypes.func.isRequired,
};
LinkButtonDialog.defaultProps = {
	linkValue: '',
};

export default LinkButtonDialog;
