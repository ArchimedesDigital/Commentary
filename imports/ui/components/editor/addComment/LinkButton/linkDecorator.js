import { CompositeDecorator } from 'draft-js';

const Link = (props) => {
	const {url} = props.contentState.getEntity(props.entityKey).getData();
	const style = {
		color: '#3b5998',
		textDecoration: 'underline',
	};
	return (
		<a href={url} style={style}>
			{props.children}
		</a>
	);
};

function findLinkEntities(contentBlock, callback, contentState) {
	contentBlock.findEntityRanges(
		(character) => {
			const entityKey = character.getEntity();

			return (
				entityKey !== null &&
				contentState.getEntity(entityKey).getType() === 'LINK'
			);
		},
		callback
	);
}

const linkDecorator = new CompositeDecorator([{
	strategy: findLinkEntities,
	component: Link,
}]);


export default linkDecorator;
export { Link, findLinkEntities };
