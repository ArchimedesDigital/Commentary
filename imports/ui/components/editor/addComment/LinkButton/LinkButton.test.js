import React from 'react';
import renderer from 'react-test-renderer';

// component:
import LinkButton from './LinkButton';

describe('LinkButton', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<LinkButton />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
