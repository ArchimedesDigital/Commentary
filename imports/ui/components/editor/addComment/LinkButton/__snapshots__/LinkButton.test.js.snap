// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<div
  className={Object {}}
  onMouseDown={[Function]}
>
  <button
    className="test"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M18.101,10.747l-3.771,3.771c-1.042,1.042-2.729,1.042-3.771,0l-1.257-1.257l1.257-1.257l1.257,1.257 c0.347,0.347,0.911,0.347,1.257,0l3.771-3.771c0.346-0.347,0.346-0.911,0-1.258l-1.257-1.257c-0.347-0.346-0.911-0.346-1.257,0 l-1.346,1.347c-0.624-0.367-1.333-0.525-2.036-0.478l2.125-2.126c1.042-1.041,2.73-1.041,3.771,0l1.257,1.257 C19.143,8.017,19.143,9.706,18.101,10.747z M10.648,15.686l-1.346,1.345c-0.347,0.347-0.911,0.346-1.257,0l-1.257-1.257 c-0.347-0.347-0.347-0.91,0-1.257l3.771-3.771c0.346-0.346,0.911-0.346,1.257,0l1.257,1.257l1.257-1.257L13.073,9.49 c-1.042-1.041-2.729-1.041-3.771,0l-3.771,3.771c-1.042,1.042-1.042,2.73,0,3.771l1.257,1.257c1.041,1.042,2.729,1.042,3.771,0 l2.125-2.125C11.982,16.211,11.275,16.053,10.648,15.686z"
        />
      </g>
    </svg>
  </button>
</div>
`;
